### DISCUSSION TITLE
Provide client API SDK for  Java  & Typescript

### DISCUSSION THREAD

**Discussion by anime777** [payloadcms/payload#3808] [Feature Requests & Ideas]
After reading the  payload compare document     [compare directus](https://payloadcms.com/compare/directus )    , I found payloadcms looks easier to maintance  compared to directus , because it control data permissions & relations via code instead of admin UI .
Do you condisider to add client SDKs to your roadmap ?  eg.  client API SDK for  Java  & Typescript  ?

**Comment by silveltman** [comment]
+1 for SDK, preferably same syntax as local API. 

I'm not very experienced with graphQL yet through, and see more and more projects move to it (ofter without SDK's).

Can graphQL the potential improve DX comparable to a SDK?

**Comment by ericuldall** [comment]
I've started actively developing one for javascript since it seems there isn't an official sdk.
I'd love some feedback: https://www.npmjs.com/package/payload-client

Once I get this base package fleshed out I'll also be working on a vue3 wrapper to allow a more reactive interface and perhaps data persistence with pinia.

If you guys like the direction and/or need any changes feel free to open issues on the github i'll be super responsive.

**Reply by ericuldall** [reply]
Just to note, it didn't make sense to emulate the local api since that has lots of options and a syntax that mostly makes sense on server. I've developed the rest sdk to be easily enumerated into service objects.

```
const User = client.service('users');
User.find()
User.get('012345678');
//etc...
```

**Reply by silveltman** [reply]
Awesome that you started working on this!

We've create our own kind of "package" inside our monorepo. Maybe it can be useful, don't know, but here it is anyway: https://stackblitz.com/edit/github-qbbbn7?file=api%2Findex.ts

**Comment by boeckMt** [comment]
I think a replacement is now [packages/sdk](https://github.com/payloadcms/payload/tree/main/packages/sdk)?

Are there some examples on how to use `packages/sdk` correctly?
I am trying to implement it as shown below, but I do not get any type information from the result.

```ts
// import { Config } from '../payload-types';

interface Post {
  id: string;
  text?: string | null;
  number?: number | null;
  updatedAt: string;
  createdAt: string;
}

interface PostsSelect<T extends boolean = true> {
  text?: T;
  number?: T;
  updatedAt?: T;
  createdAt?: T;
}

interface Config {
  collections: {
    posts: Post;
  };
  collectionsJoins: {};
  collectionsSelect: {
    posts: PostsSelect<false> | PostsSelect<true>;
  };
  db: {
    defaultIDType: string;
  };
  globals: {};
  globalsSelect: {};
  user: Post;
  auth: {};
  jobs: {
    tasks: unknown;
    workflows: unknown;
  };
}

const sdk = new PayloadSDK<Config>({
      baseInit: { credentials: 'include' },
      baseURL: `${API_HOST}/api`,
});

async function findCollection(){
    const res = await sdk.find({ collection: 'posts', select: { text: true } });
}
```
This is the type I get for res
> const res: PaginatedDocs<TransformCollectionWithSelect<T, TSlug, TSelect>>
> const test = res.docs; -> has type ` any`
